front/src/common/Map/Search/MapSearchStation.tsx

```diff
@@ -35,39 +43,43 @@ export default function MapSearchStation(props) {
 
   // Create playload based on the type of search "name" or "trigram"
 
-  const createPayload = (searchQuery) => ({
+  const createPayload = (searchQuery: (string | string[])[]) => ({
     object: 'operationalpoint',
-    query: ['and', searchQuery, ['=', ['infra_id'], infraID]],
+    query: ['and', searchQuery, ['=', ['infra_id'], infraID]] as SearchQuery,
   });
 
   // Sort on name, and on yardname
-  const orderResults = (results) =>
+  const orderResults = (results: SearchOperationalPointResult[]) =>
     results.sort((a, b) => a.name.localeCompare(b.name) || a.ch.localeCompare(b.ch));
 
   const searchByTrigrams = useCallback(async () => {
     const searchQuery = ['=i', ['trigram'], searchState];
     const payload = createPayload(searchQuery);
-    const { data, error } = await postSearch({
+    await postSearch({
       body: payload,
-    });
-    if (error) {
-      resetSearchResult();
-    } else {
-      setTrigramResults(data);
-    }
+    })
+      .unwrap()
+      .then((results) => {
+        setTrigramResults(results as SearchOperationalPointResult[]);
+      })
+      .catch(() => {
+        resetSearchResult();
+      });
   }, [searchState]);
 
   const searchByNames = useCallback(async () => {
     const searchQuery = ['search', ['name'], searchState];
     const payload = createPayload(searchQuery);
-    const { data, error } = await postSearch({
+    await postSearch({
       body: payload,
-    });
-    if (error) {
-      resetSearchResult();
-    } else {
-      setNameResults(orderResults([...data]));
-    }
+    })
+      .unwrap()
+      .then((results) => {
+        setNameResults(orderResults([...(results as SearchOperationalPointResult[])]));
```

Comment:
why not simply ?
```suggestion
        setNameResults(orderResults(results as SearchOperationalPointResult[]));
```

Reply:
Here the function orderResults mutte results and created an error if not we create a copy

Reply:
ok better make a copy in `orderResults` then

Reply:
Done